pkg/kubectl/cmd/delete.go

```diff
@@ -300,8 +314,35 @@ func (o *DeleteOptions) DeleteResult(r *resource.Result) error {
 	}
 	if found == 0 {
 		fmt.Fprintf(o.Out, "No resources found\n")
+		return nil
 	}
-	return nil
+	if !o.WaitForDeletion {
+		return nil
+	}
+	if o.DynamicClient == nil {
+		return nil
```

Comment:
The only place in code where we don't set `DynamicClient` is https://github.com/kubernetes/kubernetes/pull/64034/files#diff-7c126b9106a83157d89a336103eb3dbbR129 (apart from the unit tests).
Is that some special case? Shouldn't we always wait when `o.WaitForDeletion == true`?